src/main/java/teammates/logic/automated/EmailAction.java

```diff
@@ -93,8 +93,7 @@ public void sendEmails() {
     
     protected abstract void doPostProcessingForSuccesfulSend() throws InvalidParametersException, EntityDoesNotExistException;
     
-    protected void doPostProcessingForUnsuccesfulSend() throws EntityDoesNotExistException {
-    }
+    protected abstract void doPostProcessingForUnsuccesfulSend() throws EntityDoesNotExistException;
```

Comment:
Yup, it's used in EmailAction. PendingCommentClearedMailAction has an implementation of it.